plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/HostedMachineProviderImpl.java

```diff
@@ -210,7 +211,7 @@ protected void pullImage(CheServiceImpl service,
                                               .withBuildArgs(buildArgs),
                               progressMonitor);
 
-        } catch (ImageNotFoundException e) {
+        } catch (DockerException e) {
```

Comment:
I believe this behaves in the correct way here.
Please, take into account that this happens inside `HostedMachineProvider.pullImage`and `DockerException` can be thrown only by `DockerConnector.buildImage` method. Looking deeply, `DockerException` can be thrown from only one place where docker service response is analyzed.
According to [docker reference](https://docs.docker.com/engine/api/v1.29/#operation/ImageBuild) build endpoint could return 200, 400 and 500 error only. In our case only 500 error code is possible. Also, build is made from the following recipe: `FROM <snapshot image>`. There is no 'recipe failure'. This means, that it could be failure of: Codenvy node or network or registry. No other options. Hence:
- If the problem is on Codenvy side (i.g. not enough resources) then this should be fixed by admins and we cannot do anything here (it will fail anyway).
- If the problem is network connection or docker registry is unavailable  then fallback to the workspace recipe is the right behaviour according to spec.